pkgs/by-name/te/textpieces/package.nix

```diff
@@ -1,86 +1,67 @@
 { lib
 , stdenv
-, fetchFromGitHub
-, fetchpatch
-, python3
+, fetchFromGitLab
 , meson
 , ninja
 , pkg-config
-, vala
+, cargo
+, rustc
+, rustPlatform
+, blueprint-compiler
 , glib
 , gtk4
-, libgee
 , libadwaita
-, libportal-gtk4
-, json-glib
-, blueprint-compiler
 , gtksourceview5
-, gobject-introspection
 , wrapGAppsHook4
 , appstream-glib
 , desktop-file-utils
 }:
 
-let
-  pythonEnv = python3.withPackages (ps: with ps; [ pyyaml ]);
-in
 stdenv.mkDerivation (finalAttrs: {
   pname = "textpieces";
-  version = "3.4.1";
+  version = "4.0.6";
 
-  src = fetchFromGitHub {
+  src = fetchFromGitLab {
     owner = "liferooter";
     repo = "textpieces";
     rev = "v${finalAttrs.version}";
-    hash = "sha256-3ZUHzt3oXYgsnJVDf83JUDhcF+0DLgFfOMtpKI/FTcE=";
+    hash = "sha256-6kbGvCiaoOY+pwSmaDn1N/rbTBzEehNi/j+RI05nn6o=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "textpieces-core-1.0.0" = "sha256-HaLkL2HhH1khwsSdH64pZYtJ/WG+MLiEQPScDte/PAg=";
+    };
   };
 
   nativeBuildInputs = [
     meson
     ninja
     pkg-config
-    pythonEnv
-    vala
+    cargo
+    rustc
+    rustPlatform.cargoSetupHook
     blueprint-compiler
     wrapGAppsHook4
     appstream-glib
```

Comment:
This is [no longer used](https://gitlab.com/liferooter/textpieces/-/commit/6f9d3106c0449b9899376115f716a738df94bd38). We probably do not need to validate appstream data anyway.